File: app/src/main/java/com/riders/thelab/data/local/model/weather/WeatherData.java
package com.riders.thelab.data.local.model.weather;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Entity(tableName = "weather_data")
@Setter
@Getter
@ToString
public class WeatherData {

    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "_id")
    public int id;
    @ColumnInfo(name = "isWeatherData")
    public boolean isWeatherData;

    public WeatherData(boolean isWeatherData) {
        setWeatherData(isWeatherData);
    }
}

File: app/src/main/java/com/riders/thelab/data/local/model/weather/WeatherData.kt
package com.riders.thelab.data.local.model.weather

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "weather_data")
data class WeatherData constructor(
    @PrimaryKey
    @ColumnInfo(name = "_id")
    val id: Int = 0,

    @ColumnInfo(name = "isWeatherData")
    val isWeatherData: Boolean = false
) {
    constructor(isWeatherData: Boolean) : this(0, isWeatherData)
}
